base/parse.jl

```diff
@@ -321,14 +321,14 @@ function tryparse_internal(::Type{Complex{T}}, s::Union{String,SubString{String}
     if i₊ == i # leading ± sign
         i₊ = something(findnext(in(('+','-')), s, i₊+1), 0)
     end
-    if i₊ != 0 && s[i₊-1] in ('e','E') # exponent sign
+    if i₊ != 0 && s[prevind(s, i₊)] in ('e','E') # exponent sign
```

Comment:
Note to self: Checks like this could be sped up further if we used `codeunits(s)` and did byte comparisons rather than `Char` comparisons.  It would uglify the code and probably the speedup would be very small, however.

Reply:
(In any case, speedups to this function should go into a separate PR, not a bugfix PR, since speedups need not be backported.)